src/librbd/operation/SnapshotRemoveRequest.h

```diff
@@ -32,6 +32,12 @@ class SnapshotRemoveRequest : public Request<ImageCtxT> {
    * GET_SNAP
    *    |
    *    v (skip if unnecessary)
+   * LIST_CHILDREN
+   *    |
+   *    v (skip if unnecessary)
+   * DETACH_STALED_CHILD
```

Comment:
```suggestion
   * DETACH_STALE_CHILD
```
... and add the loop